python/ray/tune/execution/tune_controller.py

```diff
@@ -2073,15 +2085,17 @@ def _schedule_trial_restore(self, trial: Trial) -> bool:
             )
             return True
 
-        checkpoint = trial.checkpoint
+        checkpoint = trial.temporary_state.next_restore or trial.checkpoint
 
         if checkpoint.dir_or_data is None:
             logger.debug(f"Not restoring trial {trial}: No checkpoint found.")
             return False
 
         kwargs = {}
 
-        if checkpoint.storage_mode == CheckpointStorage.MEMORY:
+        if checkpoint.storage_mode == CheckpointStorage.MEMORY or isinstance(
+            checkpoint.dir_or_data, ray.ObjectRef
```

Comment:
We schedule some restores directly from the object ref

Reply:
old codepath